metricflow/plan_conversion/instance_converters.py

```diff
@@ -421,6 +450,54 @@ def transform(self, instance_set: InstanceSet) -> InstanceSet:  # noqa: D
         )
 
 
+class AliasAggregatedMeasures(InstanceSetTransform[InstanceSet]):
+    """Returns a new instance set where all measures have been assigned an alias spec"""
+
+    def __init__(self, metric_input_measure_specs: Sequence[MetricInputMeasureSpec]):
+        """Initializer stores the input specs, which contain the aliases for each measure
+
+        Note this class only works if used in conjunction with an AggregateMeasuresNode that has been generated
+        by querying a single data source for a single set of aggregated measures. This is currently enforced
+        by the structure of the DataflowPlanBuilder, which ensures each AggregateMeasuresNode corresponds to
+        a single data source set of measures for a single metric, and that these outputs will then be
+        combinded via joins.
+        """
+        self.metric_input_measure_specs = metric_input_measure_specs
+
+    def _alias_measure_instances(self, measure_instances: Tuple[MeasureInstance, ...]) -> Tuple[MeasureInstance, ...]:
+        """Update all measure instances with aliases, if any are found in the input spec set"""
+        aliased_instances: List[MeasureInstance] = []
+        aliased_input_specs = [spec for spec in self.metric_input_measure_specs if spec.alias]
+        for instance in measure_instances:
+            matches = [spec for spec in aliased_input_specs if spec.measure_spec == instance.spec]
```

Comment:
Could be good to assert that there's at most 1 match.

Reply:
I thought about this, but the reality is there could be more than 1 match because the measure_spec will be the same across all of the possibilities.

So this will work "correctly" in the end, but it may result in the wrong aliases being used.

Actually, now that I write it out, that's a disaster. I'll add the assertion for now since it should be safe with where this is called at the moment, and prioritize merging the new spec type into the base MeasureSpec after this launch sprint since that should ensure a single match in all cases.